### ISSUE TITLE
Initial value of dynamic "required" is wrong

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/react-advanced-form#113]
## What
The initial value of dynamic `required` Field prop is wrong. Once interacted with the form (means executing that dynamic fields iteration and updating) it becomes correct.

## Why
Most likely, because there is no initial dynamic fields update on construction/registration.

## How
```jsx
<Form>
            <Input
              type="email"
              name="username"
              label="Username"
              required />

            <Input
              type="password"
              label="Password"
              name="password"
              required />

            <Input
              name="firstName"
              label="First name" />
            
            <Input
              name="lastName"
              label="First name"
              required={({ fields }) => {
                return fields.firstName && !!fields.firstName.value; // this is "true" on initial load
              }} />

            <button
              className="btn btn-primary"
              type="submit">
              Create a new account
            </button>
          </Form>
```
Consider the layout above. Try to submit the form without touching anything. "lastName" field will be marked as `missing`, although it should not be required at all.

**Comment by kettanaito** [comment]
Disabled prop shouldn't be enforced by the validation, should remove.

Requires good testing.

**Comment by kettanaito** [comment]
This is related to #24, and has a high chance to be fixed under that issue.

**Comment by kettanaito** [comment]
Solved by introducing the [Reactive props](https://github.com/kettanaito/react-advanced-form/issues/24).

Merged and published under [`0.20.0`](https://github.com/kettanaito/react-advanced-form/commit/8d5832ab70c4bb1a3f337262dfbc17947aa76302). Closing.
